WooCommerce/Classes/ViewModels/InAppPurchases/UpgradesViewModel.swift

```diff
@@ -70,13 +72,13 @@ final class UpgradesViewModel: ObservableObject {
                 throw PrePurchaseError.maximumSitesUpgraded
             }
 
-            guard let plan = try await retrievePlanDetailsIfAvailable(.essentialMonthly,
-                                                                      from: wpcomPlans,
-                                                                      hardcodedPlanDataIsValid: hardcodedPlanDataIsValid)
-            else {
+            plans = try await retrievePlanDetailsIfAvailable([.essentialMonthly, .essentialYearly, .performanceMonthly, .performanceYearly],
+                                                                 from: wpcomPlans,
+                                                                 hardcodedPlanDataIsValid: hardcodedPlanDataIsValid)
```

Comment:
Do we want to hide the additional plans behind a flag at this point? Sort of:
```suggestion
var availablePlans = [AvailableInAppPurchasesWPComPlans]()
if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.freeTrialInAppPurchasesUpgradeM2) {
  availablePlans = [.essentialMonthly, .essentialYearly, .performanceMonthly, .performanceYearly]
} else {
  availablePlans = [.essentialMonthly]
}
plans = try await retrievePlanDetailsIfAvailable(availablePlans, from: wpcomPlans, hardcodedPlanDataIsValid: hardcodedPlanDataIsValid)
```

Reply:
I don't think so; the whole view is hidden behind that flag, so we can keep it simple.